### PR TITLE
OpenGL ES: reworking the use of VBO to optimize performance

### PR DESCRIPTION
**Pull request by caramelli**
This change improves the performance of the OpenGL ES port.

Nicolas Caramelli

### CODE REVIEW

File: base/OpenGLVisitor.cpp [hunk 1]
```diff
@@ -133,32 +133,29 @@ void OpenGLVisitor::visit(Group* g) {
             vector<int>::iterator indexIter = (*polyIter)->m_vIndex.begin();
             vector<int>::iterator indexEnd = (*polyIter)->m_vIndex.end();
             #ifdef HAVE_OPENGLES
-            GLuint vbo;
-            glGenBuffers(1, &vbo);
-            glBindBuffer(GL_ARRAY_BUFFER, vbo);
-            float *verts = new float[(*polyIter)->m_vIndex.size() * 5];
-            glTexCoordPointer(2, GL_FLOAT, 5 * sizeof(float), NULL);
-            glVertexPointer(3, GL_FLOAT, 5 * sizeof(float), (const float *)NULL + 2);
+            float *texcoord_pointer = new float[(*polyIter)->m_vIndex.size() * 2];
```
Comment (rzr) [inline base/OpenGLVisitor.cpp:136]: could be renamed to C++ type ie:  texccord_array or texccords

File: base/OpenGLVisitor.cpp [hunk 2]
```diff
@@ -133,32 +133,29 @@ void OpenGLVisitor::visit(Group* g) {
...
+              vertex_pointer[i * 3 + 2]   = (*shapeIter)->m_vVtxAlign[(*indexIter)].z;
             }
-            glBufferData(GL_ARRAY_BUFFER, (*polyIter)->m_vIndex.size() * 5 * sizeof(float), verts, GL_STATIC_DRAW);
             glDrawArrays(GL_TRIANGLE_FAN, 0, (*polyIter)->m_vIndex.size());
             glDisableClientState(GL_VERTEX_ARRAY);
             glDisableClientState(GL_TEXTURE_COORD_ARRAY);
-            delete verts;
-            glDeleteBuffers(1, &vbo);
+            delete vertex_pointer;
+            delete texcoord_pointer;
```
Comment (rzr) [inline base/OpenGLVisitor.cpp:158]: use delete[] for arryas

File: base/OpenGLVisitor.cpp [hunk 3]
```diff
@@ -185,32 +182,29 @@ void OpenGLVisitor::visit(Group* g) {
             vector<int>::iterator indexIter = (*polyIter)->m_vIndex.begin();
             vector<int>::iterator indexEnd = (*polyIter)->m_vIndex.end();
             #ifdef HAVE_OPENGLES
-            GLuint vbo;
-            glGenBuffers(1, &vbo);
-            glBindBuffer(GL_ARRAY_BUFFER, vbo);
-            float *verts = new float[(*polyIter)->m_vIndex.size() * 5];
-            glTexCoordPointer(2, GL_FLOAT, 5 * sizeof(float), NULL);
-            glVertexPointer(3, GL_FLOAT, 5 * sizeof(float), (const float *)NULL + 2);
+            float *texcoord_pointer = new float[(*polyIter)->m_vIndex.size() * 2];
```
Comment (rzr) [inline base/OpenGLVisitor.cpp:185]: would it make sense to have this array allocated once ? like in a userdata member of poly class , or maybe an other design pattern... I need to look back into the architecture ...

**Review by rzr — COMMENTED** [review]
looks like an improvement, forget about my comments about array naming , GL side we are talking about pointers... but delete[] operator is preferred anway

**Comment by caramelli** [comment]
OK, delete[] is now used (I'm not a C++ expert!)
About allocations, I would just like to remove the new/delete calls, the loop on indexIter, and point directly to the iterator data. But not ready, in a next commit I hope...

**Review by rzr — COMMENTED** [review]
LGTM, Is anyone against merging this?

**Comment by rzr** [comment]
for the record:

 ./helper.mk check # fps: 26.3959

and 120 fps on regular gaming with:
Version    : OpenGL ES-CM 1.1 Mesa 20.2.6

**Pull request merged**